app/src/apdu_handler.c

```diff
@@ -31,12 +31,45 @@
 #include "zxmacros.h"
 
 __Z_INLINE void handleGetPubkey(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx) {
-    extractHDPath(rx, OFFSET_DATA);
+    if (rx != 6) {
+        THROW(APDU_CODE_DATA_INVALID);
+    }
 
     uint8_t requireConfirmation = G_io_apdu_buffer[OFFSET_P1];
```

Comment:
I suggest to clarify the name of this variable:
```suggestion
    uint8_t displayAndConfirm = G_io_apdu_buffer[OFFSET_P1];
```